openshift/openshift-ansible-contrib
# [WIP] Allow different public and private DNS hostnames

#### What does this PR do?
Allow different public and private DNS hostnames.
Do not manage hostnames with openstack provider any more.
Instead, allow the cloud-init to do its job.
Ignore the hostnames for provisioned VMs. Rely on openshift-ansible
inventory variables to configure openshift clusters.

When a private FQDN differs from the public FQDN,
define 'private_fqdn' metadata for provisioned with Heat Nova servers.
Place 'openshift_(public)_ip/hostname' vars into a static and
dynamic inventory depending on the public/private IP and FQDNs setup.

#### How should this be manually tested?

* e2e with defaults should pass
* e2e with arbitrary hostnames of provisioned VMs should pass.

* Provision a cluster with `-e openshift_openstack_private_hostname_suffix: openshift-priv` and `-e openshift_openstack_private_dns_domain: example.local`.
  * The stack name, nova servers names, ansible inventory hostnames should keep using the public name prefix (defaults to cluster id, 'openshift') and belong to the public DNS domain
  * nslookup by short name executed on nodes should resolve with the public FQDN "autocompleted"
  (that depends on the hosts' search domain setup in /etc/resolv.conf, so this one is allowed to fail)
  * nslookup by public FQDNs should resolve (with a floating IP provided) via external queries against the public DNS server
  * nslookup by private FQDNs should NOT resolve via external queries against the public DNS server
  * nslookup by private FQDNs executed on nodes should resolve via internal DNS server
  * nslookup by public FQDNs executed on nodes should resolve via internal DNS server, yet with a private IP, not the floating IP

* Deploy a cluster with `-e openshift_openstack_private_hostname_suffix: openshift-priv` and `-e openshift_openstack_private_dns_domain: example.local`. E2e should pass with default and arbitrary hostnames of provisioned VMs.

* Provision using the same custom vars as above, but with an external DNS server deployed by [osp 10 ref arch](https://github.com/openshift/openshift-ansible-contrib/tree/master/reference-architecture/osp-dns) and multi-master setup. Expectations are the same as above.

Maintain the change compatible with
https://github.com/openshift/openshift-ansible/pull/6039

#### Is there a relevant Issue open for this?
Provide a link to any open issues that describe the problem you are solving.

#### Who would you like to review this?
cc: @tomassedovic @oybed PTAL
